### PR TITLE
Fixed issue #319 causing "API MISUSE" warning from CoreBluetooth

### PR DESCRIPTION
**Pull request by wojciech-kulik**
Fixed issue #319 causing "API MISUSE" warning from CoreBluetooth

### CODE REVIEW

File: Source/CharacteristicNotificationManager.swift [hunk 1]
```diff
@@ -63,6 +63,7 @@ class CharacteristicNotificationManager {
     }
 
     private func setNotifyValue(_ enabled: Bool, for characteristic: Characteristic) {
+        guard peripheral.state == .connected else { return }
```
Comment (minixT) [inline Source/CharacteristicNotificationManager.swift]: Please add logs to inform when this condition is not passed.
```suggestion
        guard peripheral.state == .connected else {
                RxBluetoothKitLog.w("""
                        \(peripheral.logDescription) is not connected. Changing notification state for not connected peripheral is not possible.
                        """)
                return 
        }
```

File: Source/Connector.swift [hunk 2]
```diff
@@ -97,7 +97,9 @@ class Connector {
                 let isConnected = strongSelf.connectedBox.read { $0.contains(peripheral.identifier) }
                 if isConnected {
                     strongSelf.disconnectingBox.write { $0.insert(peripheral.identifier) }
-                    strongSelf.centralManager.cancelPeripheralConnection(peripheral.peripheral)
+                    if strongSelf.centralManager.state == .poweredOn {
+                        strongSelf.centralManager.cancelPeripheralConnection(peripheral.peripheral)
```
Comment (minixT) [inline Source/Connector.swift]: You should call also this line `strongSelf.disconnectingBox.write { $0.insert(peripheral.identifier) }` only when bluetooth is enabled.

Comment (wojciech-kulik) [reply Source/Connector.swift]: Done

File: Source/CharacteristicNotificationManager.swift (hunk 1 above)
Comment (wojciech-kulik) [reply Source/CharacteristicNotificationManager.swift]: Done, I splitted log message into two lines, because suggested change caused swiftlint warning.

**Pull request merged**